core/execution/stoporders/stop_orders.go

```diff
@@ -107,6 +108,42 @@ func (p *Pool) Settled() []*types.StopOrder {
 	return out
 }
 
+func (p *Pool) CheckDirection(positions *positions.SnapshotEngine) []*types.StopOrder {
+	toCancel := []*types.StopOrder{}
+	for partyID, orders := range p.orders {
+		pos, ok := positions.GetPositionByPartyID(partyID)
+		if !ok {
+			continue
+		}
+		for _, order := range orders {
+			if order.SizeOverrideSetting == types.StopOrderSizeOverrideSettingPosition {
+				if pos.Size() == 0 {
```

Comment:
too many `else`'s (no need after continue, the last `else` with just an `if` also is a bit messy